Lib/test/test_codeop.py

```diff
@@ -307,14 +307,17 @@ def test_filename(self):
 
     def test_warning(self):
         # Test that the warning is only returned once.
-        with warnings_helper.check_warnings((".*literal", SyntaxWarning)) as w:
-            compile_command("0 is 0")
-            self.assertEqual(len(w.warnings), 1)
+        with warnings_helper.check_warnings(
+                (".*literal", SyntaxWarning),
+                (".*invalid", DeprecationWarning),
+                ) as w:
+            compile_command(r"'\e' is 0")
+            self.assertEqual(len(w.warnings), 2)
```

Comment:
Would it be possible to another test which only emits a single warning? For example, keep the existing "0 is 0" test, but add your new test as a new one (don't replace the existing one).

Reply:
check_warnings checks that each of the two warnings is called, so that 2 must be 1+1 and not 2+0, so the above is, in a sense, two independent tests.  We are not testing builtin compile, but that compile_command only lets 1+1 of the 3+3 warnings pass to the caller.